mods/default/chests.lua

```diff
@@ -3,16 +3,29 @@ default.chest = {}
 -- support for MT game translation.
 local S = default.get_translator
 
-function default.chest.get_chest_formspec(pos)
+function default.chest.get_chest_formspec(pos, size)
 	local spos = pos.x .. "," .. pos.y .. "," .. pos.z
-	local formspec =
-		"size[8,9]" ..
-		"list[nodemeta:" .. spos .. ";main;0,0.3;8,4;]" ..
-		"list[current_player;main;0,4.85;8,1;]" ..
-		"list[current_player;main;0,6.08;8,3;8]" ..
-		"listring[nodemeta:" .. spos .. ";main]" ..
-		"listring[current_player;main]" ..
-		default.get_hotbar_bg(0,4.85)
+	-- Find widest and highest possible combination
+	local fs_size = {
+		x = math.max(size.x, 8),
+		y = size.y + 5
+	}
+	-- Center the lists
+	local x_chest  = (fs_size.x - size.x) / 2
+	local x_player = (fs_size.x - 8) / 2
+	local formspec = (
+		"size[%i,%i]" ..
+		"list[nodemeta:%s;main;%f,0.3;%i,%i;]" ..
+		"list[current_player;main;%f,%f;8,1;]" ..
+		"list[current_player;main;%f,%f;8,3;8]" ..
+		"listring[nodemeta:%s;main]" ..
+		"listring[current_player;main]"):format(
+			fs_size.x, fs_size.y,
+			spos, x_chest, size.x, size.y,
+			x_player, size.y + 0.85,
+			x_player, size.y + 2.08,
+			spos) ..
+		default.get_hotbar_bg(x_player, 0.85 + size.y)
```

Comment:
These positions all seem rather arbitrary to me. Could this be hacky offsets to comply with the first formspec version? Please use real coordinates instead.

Reply:
Hmm, I'm not sure. I'll check it.